File: app/src/main/java/com/hannesdorfmann/search/SearchView.java
package com.hannesdorfmann.search;

import com.hannesdorfmann.mosby.mvp.MvpView;
import io.plaidapp.data.PlaidItem;
import java.util.List;

/**
 * @author Hannes Dorfmann
 */
public interface SearchView extends MvpView{

  void showLoading();

  void showContent();

  void showError(Throwable t);

  void setContentItems(List<PlaidItem> items);

  void showLoadingMore(boolean showing);

  void showLoadingMoreError(Throwable t);

  void addOlderItems(List<PlaidItem> items);

  void showSearchNotStarted();
}

File: app/src/main/java/com/hannesdorfmann/search/SearchView.kt
package com.hannesdorfmann.search

import com.hannesdorfmann.mosby.mvp.MvpView
import com.hannesdorfmann.mosby.mvp.lce.MvpLceView
import io.plaidapp.data.PlaidItem

/**
 *
 *
 * @author Hannes Dorfmann
 */
interface SearchView : MvpView {
    fun showLoading()
    fun showContent()
    fun showError(t: Throwable)
    fun setContentItems(items: List<PlaidItem>)

    fun showLoadingMore(show: Boolean)
    fun showLoadingMoreError(t: Throwable)
    fun addOlderItems(items: List<PlaidItem>)

    fun showSearchNotStarted()
}